rocAL_pybind/amd/rocal/plugin/generic.py

```diff
@@ -157,8 +157,13 @@ def __next__(self):
 
         if self.loader._name == "labelReader":
             if self.loader._one_hot_encoding == True:
-                self.loader.get_one_hot_encoded_labels(
-                    self.labels, self.device)
+                if self.device == "cpu":
+                    self.loader.get_one_hot_encoded_labels(
+                        self.labels.ctypes.data_as(
+                    ctypes.c_void_p), self.device)
+                else:
+                    self.loader.get_one_hot_encoded_labels_cupy(
```

Comment:
change to more generic name like gpu. Why this function needs to be different for cpu Vs gpu as we are passing a pointer to labels in both cases